LearnNet_CartingService/Controllers/v1/CartItemsController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Asp.Versioning;
+using FluentValidation;
+using LearnNet_CartingService.Core.DTO;
+using LearnNet_CartingService.Core.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LearnNet_CartingService.Controllers.v1
```

Comment:
I don't think you should have different controller classes here.
I suggest to create a single controller class, and just add a v2 endpoint for a specific action according to requirements.